x-pack/plugins/observability_solution/dataset_quality/public/hooks/use_dataset_quality_flyout.tsx

```diff
@@ -23,27 +22,15 @@ export const useDatasetQualityFlyout = () => {
     datasetDetails: dataStreamDetails,
     insightsTimeRange,
     breakdownField,
-  } = useSelector(service, (state) => state.context.flyout);
-  const { timeRange } = useSelector(service, (state) => state.context.filters);
-
-  const dataStreamDetailsLoading = useSelector(service, (state) =>
-    state.matches('flyout.initializing.dataStreamDetails.fetching')
-  );
-  const dataStreamSettingsLoading = useSelector(service, (state) =>
-    state.matches('flyout.initializing.dataStreamSettings.fetching')
-  );
+  } = useSelector(service, (state) => state.context.flyout) ?? {};
 
-  const datasetIntegrationsLoading = useSelector(service, (state) =>
-    state.matches('flyout.initializing.integrationDashboards.fetching')
-  );
+  const { timeRange } = useSelector(service, (state) => state.context.filters);
 
-  const loadingState = useMemo(() => {
-    return {
-      dataStreamDetailsLoading,
-      dataStreamSettingsLoading,
-      datasetIntegrationsLoading,
-    };
-  }, [dataStreamDetailsLoading, dataStreamSettingsLoading, datasetIntegrationsLoading]);
+  const loadingState = useSelector(service, (state) => ({
```

Comment:
Why do we need to keep all these ones together?
I think it's better if they are separate because they are different requests, so we can show information as soon as it arrives instead of waiting for all of them to return data

Reply:
They are still separate, just the code is simplified.